### ISSUE TITLE
understanding the relation between dimensions: input / output

### ISSUE THREAD

**Issue by chauvinSimon** [qubvel/segmentation_models.pytorch#421]
When passing RGB images of shape `(C=3, H=200, W=**630**)` to a DeepLab, masks have shapes `(n_classes, 200, **632**)`

I guess this discrepancy is due to `630` not being divisible by a certain parameter. And when changing the input shape to `(C=3, H=200, W=**632**)`, shapes get consistent.

My questions:
1) **Where in the `smp.DeepLabV3` code does a rounding / modulo operation happens that causes this discrepancy?**
2) Is there **some formula which can tell the output shape given the input shape?**

```
import segmentation_models_pytorch as smp
import torch
import torch.nn as nn


class PreTrainedSegModel(nn.Module):
    def __init__(self):
        super().__init__()

        self.model = smp.DeepLabV3(
            encoder_name='resnet18',
            classes=1,
            in_channels=3,
        )

    def forward(self, *inputs):
        return self.model.forward(*inputs)


def main():
    input_size_chw = [3, 200, 630]
    example_input_torch = torch.stack((
        torch.randn(*input_size_chw),
        torch.randn(*input_size_chw)
    ))
    if torch.cuda.is_available():
        example_input_torch = example_input_torch.cuda()

    model = PreTrainedSegModel()
    pred = model(example_input_torch)
    print(pred.shape)


if __name__ == '__main__':
    main()
```

**Comment by hhoeflin** [comment]
By the construction of semantic segmentation networks, the image input size should be a multiple of a power of 2, depending on the depth of your encoder. Here it seems that it has to be divisible by 8 in your example. 200 is divisible by 8, so the output "height" stays the same, but 630 is only divisible by 4, so the output is 632, the next number divisible by 8.
